thin-client/src/thin_client.rs

```diff
@@ -111,6 +111,7 @@ impl ClientOptimizer {
 }
 
 /// An object for querying and sending transactions to the network.
+#[deprecated(since = "1.18.4", note = "Use [RpcClient] or [TpuClient] instead.")]
```

Comment:
I don't think we should backport this to 1.18. I think either 1.19.0 or 2.0.0 would be the since version.